src/docgen/view/class.ts

```diff
@@ -31,19 +32,18 @@ export class Class {
   constructor(
     private readonly transpile: Transpile,
     private readonly klass: reflect.ClassType,
-    private readonly linkFormatter: (type: TranspiledType) => string,
   ) {
     if (klass.initializer) {
-      this.initializer = new Initializer(transpile, klass.initializer, linkFormatter);
+      this.initializer = new Initializer(transpile, klass.initializer);
     }
-    this.instanceMethods = new InstanceMethods(transpile, klass.ownMethods, linkFormatter);
-    this.staticFunctions = new StaticFunctions(transpile, klass.ownMethods, linkFormatter);
-    this.constants = new Constants(transpile, klass.ownProperties, linkFormatter);
-    this.properties = new Properties(transpile, klass.ownProperties, linkFormatter);
+    this.instanceMethods = new InstanceMethods(transpile, klass.ownMethods);
+    this.staticFunctions = new StaticFunctions(transpile, klass.ownMethods);
+    this.constants = new Constants(transpile, klass.ownProperties);
+    this.properties = new Properties(transpile, klass.ownProperties);
     this.transpiled = transpile.class(klass);
   }
 
-  public render(): Markdown {
+  public toMarkdown(linkFormatter: (type: TranspiledType) => string): Markdown {
```

Comment:
For safety's sake, shouldn't we make this a `MarkdownConversionOptions` object? (Which currently has a link formatter, but could have more in the future?

----

Or honestly, it should probably go the other way around:

```ts
class MarkdownRenderer {
  constructor(private readonly linkFormatter: LinkFormatter) {
  }

  public emitClass(klz: ClassSchema) {
    this.emit(`# ${klz.name}`); 
    for (const m of klz.instanceMethods) {
      this.emitInstanceMethod(m);
    }
    
    // ...
  }

}
```

...no?

Reply:
(Well I guess that doesn't fit with the current design very well...)

Reply:
yeah we use a lot of positional args on these, but the public API does use an object on [`Documentation.toMarkdown()`](https://github.com/cdklabs/jsii-docgen/pull/481/files#diff-850c9573c26522a8099b803d3478eaf0521a99822c5e1aeec8751de695b9d0baR207)